lahja/tools/benchmark/process.py

```diff
@@ -56,79 +61,85 @@ def start(self) -> None:
+            for n in range(config.num_events):
+                await asyncio.sleep(config.throttle)
+                await event_bus.broadcast(
+                    PerfMeasureEvent(payload, n, time.time())
+                )
 
 
 class ConsumerProcess:
 
-    def __init__(self, name: str, num_events: int) -> None:
+    def __init__(self, name: str, num_events: int, ready_event: Event) -> None:
         self._name = name
         self._num_events = num_events
         self._process: Optional[multiprocessing.Process] = None
+        self._ready_event = ready_event
 
     def start(self) -> None:
         self._process = multiprocessing.Process(
             target=self.launch,
-            args=(self._name, self._num_events)
+            args=(self._name, self._num_events, self._ready_event)
         )
         self._process.start()
 
     @staticmethod
-    async def worker(event_bus: Endpoint, num_events: int) -> None:
-        counter = itertools.count(1)
-        stats = LocalStatistic()
-        async for event in event_bus.stream(PerfMeasureEvent):
-            stats.add(RawMeasureEntry(
-                sent_at=event.sent_at,
-                received_at=time.time()
-            ))
-
-            if next(counter) == num_events:
-                event_bus.broadcast(
-                    TotalRecordedEvent(stats.crunch(event_bus.name)),
-                    BroadcastConfig(filter_endpoint=REPORTER_ENDPOINT)
-                )
-                event_bus.stop()
-                break
+    async def worker(name: str, num_events: int, ready_event: Event) -> None:
+        with Endpoint() as event_bus:
+            await event_bus.start_serving(ConnectionConfig.from_name(name))
+            await event_bus.connect_to_endpoints(ConnectionConfig.from_name(REPORTER_ENDPOINT))
+
+            ready_event.set()
+
+            counter = itertools.count(1)
+            stats = LocalStatistic()
+            async for event in event_bus.stream(PerfMeasureEvent):
+                stats.add(RawMeasureEntry(
+                    sent_at=event.sent_at,
+                    received_at=time.time()
+                ))
+
+                if next(counter) == num_events:
+                    await event_bus.broadcast(
+                        TotalRecordedEvent(stats.crunch(event_bus.name)),
+                        BroadcastConfig(filter_endpoint=REPORTER_ENDPOINT)
+                    )
+                    break
 
     @staticmethod
-    def launch(name: str, num_events: int) -> None:
-        loop = asyncio.get_event_loop()
-        event_bus = Endpoint()
-        event_bus.start_serving_nowait(ConnectionConfig.from_name(name))
-        event_bus.connect_to_endpoints_blocking(
-            ConnectionConfig.from_name(REPORTER_ENDPOINT)
-        )
+    def launch(name: str, num_events: int, ready_event: Event) -> None:
         # UNCOMMENT FOR DEBUGGING
         # logger = multiprocessing.log_to_stderr()
         # logger.setLevel(logging.INFO)
```

Comment:
Ideally we move this to the test running level so we can control it with a CLI flag from pytest.

Reply:
https://github.com/ethereum/lahja/issues/41